style/roads.mss

```diff
@@ -2321,18 +2324,22 @@ tertiary is rendered from z10 and is not included in osm_planet_roads. */
         line/line-color: @footway-fill;
         [access = 'no'] { line/line-color: @footway-fill-noaccess; }
         line/line-dasharray: 1,3;
+        [int_difficulty = 'demanding'] {line/line-dasharray: 1,12;}
         line/line-join: round;
         line/line-cap: round;
         line/line-width: @footway-width-z14;
         [zoom >= 15][int_surface = 'paved'] {
           line/line-dasharray: 2,3.5;
+          [int_difficulty = 'demanding'] {line/line-dasharray: 1,12;}
```

Comment:
Isn't this (and subsequent rules added) unnecessary?

Reply:
Wherever a line dash is specified in a filter, I'm afraid I have to 'overwrite' it with this new [int_difficulty = 'demanding'] . I could probably refactor the code in some way, but been as it is, I doubt we gain in clarity.

Reply:
You obviously would have to make the other dasharrays conditional to the  attributes they are used for.

But i agree your version makes the logic of int_difficulty overriding int_surface most explicit.